src/incremental/updateCil.ml

```diff
@@ -88,9 +89,16 @@ let update_ids (old_file: file) (ids: max_ids) (new_file: file) (changes: change
       List.iter (reset_changed_stmt (List.map snd d.unchangedNodes)) f.sallstmts;
       List.iter (assign_same_id f.sallstmts) d.unchangedNodes
   in
+  let update_var (v: varinfo) =
+    v.vid <- make_vid ()
+  in
   let reset_changed_globals (changed: changed_global) =
-    match (changed.current, changed.old) with
-    | GFun (nw, _), GFun (old, _) -> reset_changed_fun nw old changed.unchangedHeader changed.diff
+    match (changed.current.def, changed.old.def) with
+    | Some (Fun nw), Some (Fun old) -> reset_changed_fun nw old changed.unchangedHeader changed.diff
+    | Some (Var nw), Some (Var old) -> update_var nw
```

Comment:
@stilscher, @michael-schwarz  and I discussed this just now, and it turns out one would rather not call `update_var` in that case. Actually, we expect the programs to not use variables and functions that were only declared, but not defined. 

For the case of functions, we came to the conclusion that when a function previously only had a declaration and was called, and then a definition is inserted, the call sites of this now newly declared function would not be destabilized. This is because we do not record dependencies to undefined functions. 
However, as such input programs that use undefined functions and variables are malformed on a level that a compiler would reject, we decide to not introduce unnecessary handling for these cases that would clutter the implementation.